yarn-project/end-to-end/src/e2e_nft.test.ts

```diff
@@ -0,0 +1,156 @@
+import { type AccountWallet, AztecAddress, BatchCall, ExtendedNote, Fr, Note } from '@aztec/aztec.js';
+import { deriveStorageSlotInMap } from '@aztec/circuits.js/hash';
+import { pedersenHash } from '@aztec/foundation/crypto';
+import { NFTContract } from '@aztec/noir-contracts.js';
+
+import { jest } from '@jest/globals';
+
+import { setup } from './fixtures/utils.js';
+
+const TIMEOUT = 120_000;
+
+// This is a very simple test checking only the happy path. More complete tests of the NFT are implemented with TXE.
+// This test is only kept around to check that public data writes are squashed as expected.
+describe('NFT', () => {
+  jest.setTimeout(TIMEOUT);
+
+  let teardown: () => Promise<void>;
+
+  let nftContractAsAdmin: NFTContract;
+  let nftContractAsMinter: NFTContract;
+  let nftContractAsUser1: NFTContract;
+  let nftContractAsUser2: NFTContract;
```

Comment:
Do we really need this? Isn't it easier to simply connect the contract to the right account as needed?

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/8129/commits/05365233c92c77623daf546ce303cd79996960aa